doc/source/docs/user_guide/spatial_indexing.rst

```diff
@@ -0,0 +1,250 @@
+.. currentmodule:: geopandas
+
+Spatial indexing
+================
+
+When you want to know a spatial relationship (known as a spatial
+predicate) between a set of geometries A and a geometry B (or a set of
+them), you can compare geometry B against any geometry in a
+set A. However, that is not the most performant approach
+in most cases. A spatial index is a more efficient method for pre-filtering comparisons of geometries before using more computationally expensive spatial predicates.
+GeoPandas exposes the
+Sort-Tile-Recursive R-tree from shapely on any GeoDataFrame and
+GeoSeries using the :attr:`GeoSeries.sindex` property. This page outlines its options
+and common usage patterns.
+
+Note that for many operations where a spatial index provides significant
+performance benefits, GeoPandas already uses it automatically (like :meth:`~GeoDataFrame.sjoin`,
+:meth:`~GeoDataFrame.overlay`, or :meth:`~GeoDataFrame.clip`). However, more advanced use cases may require
+a direct interaction with the index.
```

Comment:
Maybe we could mention the many-to-many case up front as an example of a situation where direct interaction makes sense?

Reply:
Not sure I follow what you meant...